### PR TITLE
feat(CI): add a GitHub Action to lint the YAML files

### PR DESCRIPTION
**Pull request by jbampton**
Lint some YAML
Add a `.yamllint` config file

### What this PR does / why we need it:
Standardizes our YAML files

### Pre-submission checklist:

* [X] Did you explain what problem does this PR solve? Or what new features have been added?
* [ ] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [ ] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

File: conf/apisix.yaml [hunk 1]
```diff
@@ -19,7 +19,7 @@ routes:
...
     uri: /hello
     upstream:
-        nodes:
-            "127.0.0.1:1980": 1
-        type: roundrobin
-#END
+      nodes:
+        "127.0.0.1:1980": 1
+      type: roundrobin
+# END
```
Comment (juzhiyuan) [inline conf/apisix.yaml:25]: Not sure, cc @spacewander

File: conf/debug.yaml [hunk 2]
```diff
@@ -28,4 +28,4 @@ hook_phase:                     # module and function list, name: hook_phase
     - http_body_filter_phase
     - http_log_phase
 
-#END
+# END
```
Comment (juzhiyuan) [inline conf/debug.yaml:31]: ,

File: kubernetes/service-monitor-for-prometheus.yaml [hunk 3]
```diff
@@ -36,4 +36,3 @@ spec:
       app: apisix-gw
   namespaceSelector:
     any: true
-
```
Comment (juzhiyuan) [inline kubernetes/service-monitor-for-prometheus.yaml:39]: 🤔 May need to keep the last line?

Comment (jbampton) [reply kubernetes/service-monitor-for-prometheus.yaml:39]: Last line is there seen here:

view-source:https://raw.githubusercontent.com/apache/apisix/9f771b29232929dc5e85b9c97c4d43ff86924ee5/kubernetes/service-monitor-for-prometheus.yaml

That file has 2 lines at the end

File: conf/apisix.yaml (hunk 1 above)
Comment (spacewander) [inline conf/apisix.yaml:25]: The `#END` is a special EOF token. Can we exclude it from the lint?

**Review by spacewander — CHANGES_REQUESTED** [review]

File: conf/config-default.yaml [hunk 4]
```diff
@@ -205,8 +206,8 @@ plugins:                          # plugin list (sorted in alphabetical order)
   - consumer-restriction
   - cors
   - echo
-  #- error-log-logger
-  #- example-plugin
+  # - error-log-logger
+  # - example-plugin
```
Comment (spacewander) [inline conf/config-default.yaml:210]: Need to update this line: https://github.com/apache/apisix/blob/f476d2c3dc68439406a9531fc8b4a2a2d50c1333/t/APISIX.pm#L75

**Review by spacewander — CHANGES_REQUESTED** [review]

**Review by spacewander — APPROVED** [review]

**Review by moonming — APPROVED** [review]

**Pull request merged**